test/aio5gc/build.go

```diff
@@ -25,8 +25,8 @@ func (f *FiveGCBuilder) WithConfig(conf config.Config) *FiveGCBuilder {
 	return f
 }
 
-func (f *FiveGCBuilder) WithNASDispatcherHook(hook func(*nas.Message, *context.UEContext, *context.GNBContext, *context.Aio5gc) (bool, error)) *FiveGCBuilder {
-	f.nasHook = append(f.nasHook, hook)
+func (f *FiveGCBuilder) WithNASDispatcherHook(hooks map[uint8]func(*nas.Message, *context.UEContext, *context.GNBContext, *context.Aio5gc) (bool, error)) *FiveGCBuilder {
```

Comment:
Since you are using the builder pattern, I feel it would make more sense to be able to add one hook at a time:
```go
WithNASDispatcherHook(msgType uint8, hook func(*nas.Message, *context.UEContext, *context.GNBContext, *context.Aio5gc) (bool, error))
```